alphastats/llm/llm_integration.py

```diff
@@ -289,6 +289,21 @@ def get_print_view(self, show_all=False) -> List[Dict[str, Any]]:
             )
         return print_view
 
+    def get_chat_log_txt(self) -> str:
+        """Get a chat log in text format for saving. It excludes tool replies, as they are usually also represented in the artifacts."""
+        messages = self.get_print_view(show_all=True)
+        chatlog = ""
+        for message in messages:
+            if message["role"] == "tool":
+                continue
+            chatlog += f"{message['role'].capitalize()}: {message['content']}\n"
+            if len(message["artifacts"]) > 0:
+                chatlog += "-----\n"
+            for artifact in message["artifacts"]:
+                chatlog += f"Artifact: {artifact}\n"
```

Comment:
would be cool if e.g. plotly plots could be actually displayed by copying this string (might need some escaping)

Reply:
This is possible with some alterations to the outputted string as it is now. After trying I identified these points that need changing, but this list is likely incomplete and automating it could be detrimental (as I think ... also gets introduced to the data arrays if they become too long):
- remove any lines with ..., which includes hovertemplates and the layout template (therefore becomes a bit uglier)
- add quotation marks to strings in axis category arrays

Then one can reproduce the figure with this code, copying the curly bracket from the artifact output.
```
import plotly.graph_objects as go
from numpy import array
go.Figure(**{
    'data': ...,
    'layout': ...
})
```

This could be mitigated by exporting the figures to html, but then I would suggest one file per figure.

Reply:
hm, seems quite brittle indeed .. then just leave it for now